src/clients/api/mutations/supplyBnb.ts

```diff
@@ -0,0 +1,34 @@
+import Web3 from 'web3';
+import type { TransactionReceipt } from 'web3-core';
+import { VBnbToken } from 'types/contracts';
+import { getVBepToken } from 'utilities';
+
+export interface ISupplyBnbInput {
+  tokenContract: VBnbToken;
+  web3: Web3;
+  account: string | undefined;
+  amount: string;
+}
+
+const vBnbAddress = getVBepToken('bnb').address;
+
+export type SupplyBnbOutput = TransactionReceipt;
+
+const supplyBnb = async ({
+  web3,
+  tokenContract,
+  account,
+  amount,
+}: ISupplyBnbInput): Promise<SupplyBnbOutput> => {
+  const contractData = tokenContract.methods.mint().encodeABI();
+  const tx = {
```

Comment:
That's a small thing, but given you're passing a `web3` instance you could grab the contract here directly (so you wouldn't have to pass the contract too).